### ISSUE TITLE
How do I initiate playing of song on click of song URL?

### ISSUE THREAD

**Issue by Oliverq755** [mudigal-technologies/ngx-audio-player#42]
I found this library very useful however when I started the implementation in my application I lost how to implement the play action on click of URL dynamically.
## Use case
I have multiple cards of some information like (songName, songURL, scope, rating) like 
```json
{
 "songName":"The Dark Kight Theme",
 "songURL":"http://oliverspost.com/Batman%20The%20Dark%20Knight%20Theme%20-%20Hans%20Zimmer.mp3",
 "scope":"Online",
 "Rating":4.5
}
```

I have multiple cards like this and a play button on it. When I click it I want to initiate the play on the ngx-audio-player component. [SCREENSHOT ATTACHED]
Any help and ideas will be appreciated. 
![Screenshot from 2020-02-18 23-07-48](https://user-images.githubusercontent.com/34798266/74762319-9e778000-52a3-11ea-8077-1420b43f8a4e.png)

**Comment by raphael-jungers** [comment]
You could try like this :

Retrieve your player inside your TypeScript component

`@ViewChild(MatBasicAudioPlayerComponent)
    player: MatBasicAudioPlayerComponent;`

When a card is clicked, change the audioUrl value, reset the player and play the song

`play(trackInfo: Track): void {
    this.currentAudioUrl = trackInfo.songURL;
    this.player.resetSong();
    this.player.play();
  }`

**Comment by Oliverq755** [comment]
@raphael-jungers 
I tried with the solution but didn't work out. I am getting error like 
`ERROR TypeError: Cannot read property 'resetSong' of undefined
    at ....
ERROR TypeError: Cannot read property 'play' of undefined
    at... `
``

**Comment by vmudigal** [comment]
Available in "ngx-audio-player" starting version v12.0.1

HTML
```
<div>
    <button (click)="play()" mat-stroked-button>Play</button>
</div>
<div>
    <button (click)="pause()" mat-stroked-button>Pause</button>
</div>
<div>
    <button (click)="stop()" mat-stroked-button>Stop</button>
</div>

<ngx-audio-player #player [playlist]="singleTrack">
</ngx-audio-player>
```

TS

```
  @ViewChild('player', { static: false })
  simplePlayer: AudioPlayerComponent;

  // Single
  singleTrack: Track[] = [
    {
        title: 'Audio Title',
        link: 'Audio URL',
        duration: 225,
        artist: 'Artist'
    }
];

  play() {
      this.simplePlayer.play();
  }

  pause() {
      this.simplePlayer.pause();
  }

  stop() {
      this.simplePlayer.stop();
  }
```
